### ISSUE TITLE
Errors encountered in run-tests.sh

### ISSUE THREAD

**Issue by mdecesar** [nanograv/PINT#248]
Hi,

I pulled the PINT repository and reran "python setup.py build_ext --inplace" and run-tests.sh.  I got a bunch of errors that I hadn't gotten before pulling the latest updates (before pulling, I had no errors when running the tests).  I removed and re-cloned the repository, thinking that might be cleaner, but this resulted in the same errors.  Hoping someone can help me get rid of these errors and finish the installation.

To install PINT, I did:

git clone https://github.com/nanograv/PINT.git
cd PINT
python setup.py install build_ext --inplace
cd tests
./run-tests.sh

The errors I encountered in run-tests are the following.  (I didn't get any errors from setup.py.)

(1)  For one error, it's clear I need to be able to import fftfit, but I'm not able to find fftfit in pip.  I see there's an fftfit in presto; is this the same as what I need for PINT?  Do I need to install presto, or only fftfit?  The error is:

ERROR: Failure: ImportError (No module named fftfit)
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/nose/loader.py", line 418, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/local/lib/python2.7/site-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/local/lib/python2.7/site-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/Users/decesame/Software/Pulsar_Software/PINT/tests/test_event_optimize-DISABLED.py", line 9, in <module>
    import pint.scripts.event_optimize as event_optimize
  File "/usr/local/lib/python2.7/site-packages/pint-0.5.2+10.g3d32ace-py2.7-macosx-10.11-x86_64.egg/pint/scripts/event_optimize.py", line 15, in <module>
    import sys, os, copy, fftfit
ImportError: No module named fftfit

(2)  Another error is an assertion error:
FAIL: test_from_dir (test_solar_system_body.TestSolarSystemDynamic)
Traceback (most recent call last):
  File "/Users/decesame/Software/Pulsar_Software/PINT/tests/test_solar_system_body.py", line 67, in test_from_dir
    assert solar_system_ephemeris._value == path
AssertionError

(3)  The third error repeats many times for different tests.  I'm having trouble deciphering what it actually means.  It has something to do with str2ld.  The str2ld files are there in pint/cutils, but I'm currently not able to "import str2ld".  Should I be able to import str2ld if I have the files in pint/cutils, or do I have to do something further?

Here is an example of the str2ld error:
ERROR: test_result (test_zima.TestZima)
Traceback (most recent call last):
  File "/Users/decesame/Software/Pulsar_Software/PINT/tests/test_zima.py", line 18, in test_result
    zima.main(cmd.split())
  File "/usr/local/lib/python2.7/site-packages/pint-0.5.2+10.g3d32ace-py2.7-macosx-10.11-x86_64.egg/pint/scripts/zima.py", line 39, in main
    m = pint.models.get_model(args.parfile)
  File "/usr/local/lib/python2.7/site-packages/pint-0.5.2+10.g3d32ace-py2.7-macosx-10.11-x86_64.egg/pint/models/model_builder.py", line 258, in get_model
    mb = model_builder(name,parfile)
  File "/usr/local/lib/python2.7/site-packages/pint-0.5.2+10.g3d32ace-py2.7-macosx-10.11-x86_64.egg/pint/models/model_builder.py", line 88, in __init__
    self.get_model_instance(self.parfile)
  File "/usr/local/lib/python2.7/site-packages/pint-0.5.2+10.g3d32ace-py2.7-macosx-10.11-x86_64.egg/pint/models/model_builder.py", line 241, in get_model_instance
    self.model_instance.read_parfile(parfile)
  File "/usr/local/lib/python2.7/site-packages/pint-0.5.2+10.g3d32ace-py2.7-macosx-10.11-x86_64.egg/pint/models/timing_model.py", line 605, in read_parfile
    if getattr(self, par).from_parfile_line(l):
  File "/usr/local/lib/python2.7/site-packages/pint-0.5.2+10.g3d32ace-py2.7-macosx-10.11-x86_64.egg/pint/models/parameter.py", line 341, in from_parfile_line_regular
    self.set(k[1])
  File "/usr/local/lib/python2.7/site-packages/pint-0.5.2+10.g3d32ace-py2.7-macosx-10.11-x86_64.egg/pint/models/parameter.py", line 293, in set
    self.value = value
  File "/usr/local/lib/python2.7/site-packages/pint-0.5.2+10.g3d32ace-py2.7-macosx-10.11-x86_64.egg/pint/models/parameter.py", line 222, in value
    self._quantity = self.set_quantity(val)
  File "/usr/local/lib/python2.7/site-packages/pint-0.5.2+10.g3d32ace-py2.7-macosx-10.11-x86_64.egg/pint/models/parameter.py", line 476, in set_quantity_float
    result = setfunc_no_unit(val)
  File "/usr/local/lib/python2.7/site-packages/pint-0.5.2+10.g3d32ace-py2.7-macosx-10.11-x86_64.egg/pint/models/parameter.py", line 468, in <lambda>
    setfunc_no_unit = lambda x:  data2longdouble(x) * self.units
  File "/usr/local/lib/python2.7/site-packages/pint-0.5.2+10.g3d32ace-py2.7-macosx-10.11-x86_64.egg/pint/utils.py", line 358, in data2longdouble
    return str2longdouble(data)
  File "/usr/local/lib/python2.7/site-packages/pint-0.5.2+10.g3d32ace-py2.7-macosx-10.11-x86_64.egg/pint/utils.py", line 314, in str2longdouble
    return str2ldarr1(input_str)[0]
TypeError: 'NoneType' object has no attribute '\_\_getitem\_\_'

Thanks,
Megan

**Comment by jellis18** [comment]
Hi Megan,

The build and tests have changed somewhat. You should just be able to do

make test

You may have to do pip install requirements_dev.txt first. I don't think we have to do the inlace builds anymore. This reminds me that we need to update the README and docs.

Thanks

Justin

-------------------------------------------------------------------------------
Justin A. Ellis
Einstein Postdoctoral Fellow
Jet Propulsion Laboratory, California Institute of Technology
Pasadena CA, 91109

**Comment by paulray** [comment]
For error (1), it is because that test requires fftfit from PRESTO.  It is disabled in our automated testing for that reason (thus the -DISABLED) in the name. Soon, we'll put a pure python version of FFTFIT into PINT and that will go away.

I have no idea about error (2). I've not seen that before.

For error (3), I believe it is because you installed in place then changed directories, so you aren't running the in place version of PINT, but rather some installed PINT.  See "Running Tests" on the Wiki:
https://github.com/nanograv/PINT/wiki/Installation-Notes
Basically try it again from your PINT dir (not in PINT/tests), and run `run-tests-local.sh`

But, as @jellis18 said, we are in a transition so there is a new way of running tests and we no longer recommend installing in place.  We're moving the docs and will be updating them shortly.

**Comment by mdecesar** [comment]
Hi Justin and Paul, thanks a lot.  Using "pip install -r requirements_dev.txt" and then "make test" worked for me.

In running "make test," I got one warning that looks similar to something Paul had mentioned in another post:
WARNING: ErfaWarning: ERFA function "dtf2d" yielded 1 of "dubious year (Note 6)" [astropy._erfa.core]

Also a future warning:
/usr/local/lib/python2.7/site-packages/astropy/table/column.py:264: FutureWarning: comparison to `None` will result in an elementwise object comparison in the future.
  return self.data.__eq__(other)

There are also many unrecognized parfile lines, but that's normal.

Megan

**Comment by paulray** [comment]
Cool. So I guess the action to close this issue is to remove the run-tests scripts and update the documentation, which should be moved off the wiki and into our new docs pages.

**Comment by jellis18** [comment]
Agreed

**Comment by paulray** [comment]
Done in PR #252
